### PR TITLE
fix(extra.lang.angular): treesitter not enabled for angular templates

### PR DESCRIPTION
**Pull request by mishasbtr**
## Problem
Treesitter was not working in angular html templates after enabling `lang.angular` extra.

## Solution
Added a change recommeded by angular treesitter developer to enable treesitter for components and containers.

### Reference to recommendation in treesitter docs
[Reference to the recommendation for this hack](https://github.com/dlvandenberg/tree-sitter-angular?tab=readme-ov-file#filetype).

### CODE REVIEW

File: lua/lazyvim/plugins/extras/lang/angular.lua [hunk 1]
```diff
@@ -14,6 +14,18 @@ return {
...
+        pattern = {
+          [".*%.component%.html"] = "angular.html",
+          [".*%.container%.html"] = "angular.html",
+        },
+      })
+      vim.api.nvim_create_autocmd("FileType", {
+        pattern = "angular.html",
+        callback = function()
+          vim.treesitter.language.register("angular", "angular.html")
+        end,
```
Comment (folke) [inline lua/lazyvim/plugins/extras/lang/angular.lua:27]: IS this really needed to do in an autocmd?
Why not just register it?

Comment (mishasbtr) [reply lua/lazyvim/plugins/extras/lang/angular.lua]: Indeed, you are right, this is not necessary.
Removed the wrapping of registration with autocmd.

**Comment by folke** [comment]
Thanks!

**Comment by BBboy01** [comment]
is your lsp still working after setting the filetype to `angular.html`

**Comment by mishasbtr** [comment]
> is your lsp still working after setting the filetype to `angular.html`

Thank you for catching that! Indeed, lsp stops working after changing the filetype.
Working on a fix.

**Comment by mishasbtr** [comment]
@BBboy01 
PR with a fix: https://github.com/LazyVim/LazyVim/pull/3485

**Comment by mishasbtr** [comment]
Re-fixed in https://github.com/LazyVim/LazyVim/pull/3488

**Pull request merged**